server/src/main/java/com/objectcomputing/checkins/services/questions/QuestionController.java

```diff
@@ -24,8 +35,26 @@
     @Inject
     private QuestionServices questionService;
 
-    public void setQuestionService(QuestionServices questionService) {
-        this.questionService = questionService;
+    public QuestionController(QuestionServices questionServices) {
+        this.questionService = questionServices;
+    }
+
+    @Error(exception = QuestionNotFoundException.class)
```

Comment:
@mkimberlin 

I think we need a story later to make a global error handler and retrofit the services.  

I wrote my original generic badargexception for my story because I didn't want to step on any toes. John is going in a direction I originally wanted to take this which was to make more than just one handler and to return more than just a bad request for everything.

I like the idea of these and think they should be brought up to the global level. It wouldn't take much effort with Intellij and it's ability to refactor.

Reply:
Write that puppy up! (you almost just did) 😂